src/Http/Http/src/Builder/ApplicationBuilder.cs

```diff
@@ -99,9 +99,9 @@ public RequestDelegate Build()
                 return Task.CompletedTask;
             };
 
-            foreach (var component in _components.Reverse())
+            for (var c = _components.Count; c-- > 0; )
```

Comment:
```suggestion
            for (var c = _components.Count; c > 0; c--)
```

See https://github.com/dotnet/aspnetcore/pull/28393#discussion_r536027865 if I care about it or not...

Reply:
Say what?

Reply:
Did you try your change with a debugger? 
If `c = _components.Count` the indexed access `_components[c]` will throw a IndexOutOfRangeException, as the valid indexes go from `0` to `_components.Count - 1`.

Reply:
> Did you try your change with a debugger?

Did you?

Check it out in [sharplab.io](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQcAMACOSAsA3OuogMzoDe62tuCCNd1adb2AbgIZjYA2ASwDOwbAF5sUAKYB3bABlhwADx5MAPibtsLbdoBEAe2n6ANFr219wGYbMXL1gBZgpU+6z0BfYp+0O6ADNDXgAKbl4BcX4lADoAYUNoYEJsAQBadOx1bExUgEoA2l1LOjwATlDBEQBtAQBdfN9LLwtWtC8gA=).

Reply:
Ah, I'm totally confused by this (untypical) loop and didn't consider that the condition is checked before the body is executed.
Another reason why I wouldn't write the loop that way -- it's just not usual here (albeit correct, and hence sorry for that).

Reply:
This is legal and by-design in C#. The fact that you didn't know or use it doesn't make it untypical.

Reply:
See https://github.com/dotnet/aspnetcore/pull/28393#discussion_r536314228 for style guidance.

Reply:
Done! Sorry for the delay.